util/argo/argo.go

```diff
@@ -301,6 +301,20 @@ func ValidateDestination(ctx context.Context, dest *argoappv1.ApplicationDestina
 	return nil
 }
 
+func SetDestinationServer(ctx context.Context, dest *argoappv1.ApplicationDestination, db db.ArgoDB) error {
```

Comment:
What is the difference between this method and `ValidateDestination` ? I think it is pretty much the same. Cannot we use `ValidateDestination` instead?

Reply:
Not much, `ValidateDestination` has an additional validation like
https://github.com/argoproj/argo-cd/blob/c147320687c7bf0e3da885c9ed02fbfcb6b45c2c/util/argo/argo.go#L296

We can use `ValidateDestination` as well, which means `ValidateDestination` will move out of `if validate {}` check. I don't know why it was added to 
https://github.com/argoproj/argo-cd/blob/c147320687c7bf0e3da885c9ed02fbfcb6b45c2c/server/application/application.go#L752